k2/csrc/array_ops.cu

```diff
@@ -147,63 +148,55 @@ Array1<int32_t> SpliceRowSplits(int32_t num_arrays,
 
 bool ValidateRowIds(Array1<int32_t> &row_ids, Array1<int32_t> *temp) {
   ContextPtr ctx = row_ids.Context();
-  int32_t *data = row_ids.Data();
+  const int32_t *data = row_ids.Data();
   int32_t dim = row_ids.Dim();
-  if (dim == 0) return true;  // will treat this as valid.a
-
-  if (ctx->GetDeviceType() == kCpu) {
-    if (data[0] < 0) return false;
-    for (int32_t i = 0; i + 1 < dim; i++)
-      if (data[i] > data[i + 1]) return false;
-  }
+  if (dim == 0) return true;  // will treat this as valid
+  // note `row_ids[0]` may copy memory from device to host
+  if (row_ids[0] < 0) return false;
 
   Array1<int32_t> temp_array;
   if (temp == nullptr || temp->Dim() == 0) {
-    temp_array = Array1<int32_t>(row_ids.Context(), 1);
-    temp = &temp_array;
+    temp_array = Array1<int32_t>(ctx, 1);
+  } else {
+    temp_array = temp->Range(0, 1);
   }
-
+  temp = &temp_array;
   *temp = 0;
```

Comment:
For the following use case
```cpp
Array1<int32_t> row_ids = ...; // a non-empty row_ids
Array1<int32_t> temp;
ValidateRowIds(row_ids, &temp);
```

After returning from `ValidateRowIds`, `temp` is still empty. I don't see the need for passing `temp`
as an argument. It can be defined inside the function.

Reply:
Will not fix, see my explanation above